ingestion/src/metadata/ingestion/source/dashboard/quicksight/metadata.py

```diff
@@ -161,61 +186,97 @@ def yield_dashboard_chart(
-                )
-                yield self._get_add_lineage_request(
-                    to_entity=to_entity, from_entity=from_entity
-                )
+                    )["DataSet"]["PhysicalTableMap"].values()
+                ):
+                    if any(
+                        data_source_resp in data_source
+                        for data_source_resp in ["S3Source", "CustomSql"]
+                    ):
+                        logger.warning(
+                            "We currently don't support lineage to S3 Source or Custom Sql Queries"
+                        )
+                        continue
+
+                    # db_name = data_source
+                    schema_name = data_source["RelationalTable"]["Schema"]
+                    table_name = data_source["RelationalTable"]["Name"]
+
+                    list_data_source_func = lambda kwargs: self.client.list_data_sources(  # pylint: disable=unnecessary-lambda-assignment
+                        **kwargs
+                    )
+
+                    data_source_summary_list = self._check_pagination(
+                        listing_method=list_data_source_func,
+                        entity_key="DataSources",
+                    )
+
+                    data_source_ids = [
+                        data_source_arn["DataSourceId"]
+                        for data_source_arn in data_source_summary_list
+                        if data_source_arn["Arn"]
+                        in data_source["RelationalTable"]["DataSourceArn"]
+                    ]
+
+                    for data_source_id in data_source_ids:
+                        data_source_dict = self.client.describe_data_source(
+                            AwsAccountId=self.aws_account_id,
+                            DataSourceId=data_source_id,
+                        )["DataSource"]["DataSourceParameters"]
+                        for db in data_source_dict.keys():
+                            from_fqn = None
+                            try:
+                                database_name = data_source_dict[db]["Database"]
+                            except Exception as err:
+                                logger.error(err)
+                                logger.debug(traceback.format_exc())
+                                database_name = None
+                                from_fqn = fqn.build(
+                                    self.metadata,
+                                    entity_type=Table,
+                                    service_name=db_service_name,
+                                    table_name=table_name,
+                                    schema_name=schema_name,
+                                    database_name="",
+                                )
+                            from_entity = self.metadata.get_by_name(
+                                entity=Table,
+                                fqn=f"{db_service_name}{fqn.FQN_SEPARATOR}"
```

Comment:
@ayush-shah any reason for not using the `from_fqn` here 🤔

Reply:
if we are getting the whole fqn: service name, db, schema and table, we can avoid the es calls altogether..
i have added from_fqn as a safe fallback if we are not getting any db name, we do the es calls and fetch the full fqn

Reply:
Maybe you can take advantage of [this](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/fqn.py#L135) parameter to skip the es calls wherever required.

Building the fqn via fqn builder is important since it takes care of [quotes](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/fqn.py#L81), etc in the names

Reply:
yes, adding that as well